### ISSUE TITLE
Allow running `@QuarkusMainIntegrationTest` on non short-lived apps

### ISSUE THREAD

**Issue by cohenor** [quarkusio/quarkus#26768]
### Description

Hey,

My colleagues and I working on a Quarkus application that basically runs a server that reads data from a Unix domain socket over UDP and performs some BL on top of the messages it receives.

To run the socket server we're providing a custom main method with `@QuarkusMain` - 
```kotlin
// SocketServerMain.kt
@QuarkusMain
object SocketServerMain {
    @JvmStatic
    fun main(args: Array<String>) {
        Quarkus.run(SocketServer::class.java, *args)
    }
}
```

```kotlin
// SocketServer.kt
class SocketServer() : QuarkusApplication {
    override fun run(vararg args: String?): Int {
        start()

        return 0
    }

    private fun start() {
        // Socket server main loop
    }
}
```

While running and testing the application manually, we had no issues and everything worked just fine.

We did bump into an issue when trying to write integration tests for the server - it seemed as though the server never really started running when we used `@QuarkusMainIntegrationTest`.
After some digging we've found out that `@QuarkusMainIntegrationTest` is meant for working short lived applications (i.e command line apps) and waits for the process completion before actually invoking the test suite.

We would like to have an option to run integration tests on a custom `main` application without blocking the tests until the application's exit code is returned - similarly to how it's done with HTTP servers for example.

### Implementation ideas

To overcome our issue with running the ITs - we've created a custom `SocketServerMainIntegrationTestExtension` for junit, which we hacked to work with some package private functions from [IntegrationTestUtil](https://github.com/quarkusio/quarkus/blob/main/test-framework/junit5/src/main/java/io/quarkus/test/junit/IntegrationTestUtil.java).
The main difference from the `QuarkusMainIntegrationTestExtension` is that it calls `IntegrationTestUtil.startLauncher` to run the application rather than [`launcher.runToCompletion`](https://github.com/quarkusio/quarkus/blob/main/test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusMainIntegrationTestExtension.java#L158).

I'm not sure that this is the optimal way to go, and perhaps our solution won't cover all integration tests use cases, but for our rather small app, it actually did the trick.
One limitation we did notice is that the log level of the running application must not be `ERROR`/`WARN` - otherwise it seems as though Quarkus does not detect any activity from the application under test and assumes it does not behave as expected 🤔

**Comment by geoand** [comment]
I'd say, since you already have something, maybe open a draft PR so we can discuss

**Comment by cohenor** [comment]
> I'd say, since you already have something, maybe open a draft PR so we can discuss

Sounds good, will do 👍

**Comment by geoand** [comment]
@cohenor just checking in, were you able to come up with the change you had in mind?

**Comment by cohenor** [comment]
Hey @geoand, sorry for the late response, I went on a long vacation and didn't get to chance to open a pull request in this repository.
I do have [this change in a local fork](https://github.com/cohenor/quarkus/pull/1/files) though in case you want to take a look.
It's a **rough draft** and TBH it's the first time I'm digging into Quarkus' core and internals so there might be some mistakes here and there. But it was working beautifully in my original project.

**Comment by geoand** [comment]
Thanks! I'll have a look soon.

**Comment by geoand** [comment]
Quick questions: What is the difference between the work you have done in `QuarkusMainServerIntegrationTest` and `QuarkusIntegrationTest`?

**Comment by cohenor** [comment]
When we tried using `QuarkusIntegrationTest` for the UNIX socket server, we found out the the custom main we added was not called, and it seems like this happens because `QuarkusIntegrationTest` is assuming the application is an HTTP server and is attempting to run it rather than our custom `@QuarkusMain`.

Then I bumped into `QuarkusMainIntegrationTest` for command mode apps, which seemed like what I was looking for, except for the fact that it blocks the test suite until the launcher finishes.
In that sense, when adding `QuarkusMainServerIntegrationTest`, what I had in mind is to be able to run the UNIX server as if it was a command and allow the test suite to run against the UNIX server.
